pallets/randomness/src/types.rs

```diff
@@ -89,15 +89,12 @@ impl<Hash: Clone> RandomnessResult<Hash> {
 		}
 	}
 	/// Increment request count
-	pub fn increment_request_count<T: Config>(&self) -> Result<Self, DispatchError> {
-		let new_request_count = self
-			.request_count
-			.checked_add(1u64)
-			.ok_or(Error::<T>::RequestCounterOverflowed)?;
-		Ok(RandomnessResult {
+	pub fn increment_request_count<T: Config>(&self) -> Self {
+		let new_request_count = self.request_count.saturating_add(1u64);
```

Comment:
Would this mean that in the case of an overflow that we would start overwriting requests? That would be all sorts of problematic.

Maybe this is an acceptable place where we could just panic...

Reply:
> Would this mean that in the case of an overflow that we would start overwriting requests? That would be all sorts of problematic.

We are overwriting the request count, the number of requests for this specific randomness result. 

This also only happens if there are more than `u64::max()` requests for a given randomness result.

But no, we are not overwriting any given request.

Reply:
Ah, yes, I forgot what its purpose is. Hitting this case (as its currently written) would mean we have a `RandomnessResult` that would look like it could be removed before it actually should be.

But this is certainly impossible because we couldn't fit enough transactions in blocks...